ouroboros-consensus/test-consensus/Test/Consensus/ChainSyncClient.hs

```diff
@@ -309,7 +309,7 @@ runChainSync securityParam maxClockSkew (ClientUpdates clientUpdates)
     -- STM variable to record the final synched candidate chain
     varCandidateChain <- atomically $ newTVar Nothing
 
-    onLaterSlot btime (finalSlot numSlots) $ do
+    onLaterSlot btime (slotAfterLast numSlots) $ do
```

Comment:
When I first read this, I worried that this would never be triggered as `testBlockchainTime` only ticks for `numSlots` slots, so it would not tick at the slot after the last. The name `finalSlot` did not evoke this worry. Maybe combine `onLaterSlot` with `slotAfterSlot` into `onEnd` (or something)?

Reply:
We chatted on Slack -- I'm going to comb the details more carefully. Especially, e.g. `testBlockchainTime`, which seems to create one "additional" slot (the one referred to by `finalSlot`/`slotAfterLast`). My initial theory: either `testBlockchainTime` and `finalSlot` have an off-by-one bug, or `NumSlots` is misnamed.

Reply:
In the mean-time, I've removed that commit in order to expedite this PR.